resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Constants.scala

```diff
@@ -65,11 +65,13 @@ private[spark] object Constants {
   val ENV_CLASSPATH = "SPARK_CLASSPATH"
   val ENV_DRIVER_BIND_ADDRESS = "SPARK_DRIVER_BIND_ADDRESS"
   val ENV_SPARK_CONF_DIR = "SPARK_CONF_DIR"
+  val ENV_SPARK_USER = "SPARK_USER"
```

Comment:
I guess this is for setting the correct user. But I think hadoop libs should pick the correct user in SparkContext where `Utils.getCurrentUserName()` is used. In addition I think we should allow the pods to run with any username (eg. with customized images) and we should have a security context per container like:  `SecurityContextBuilder().withRunAsUser()`. In the latter scenario hadoop libraries will pick that logged in user as well.